src/childinfo.c

```diff
@@ -63,54 +58,38 @@ void closeChildInfoPipe(void) {
     }
 }
 
-/* Send save data to parent. */
-void sendChildInfo(int information_type, size_t info) {
-    if (server.child_info_pipe[1] == -1) return;
-
-    child_info_data buffer = {.information_type = information_type, .info = info};
-    ssize_t wlen = sizeof(buffer);
-
-    if (write(server.child_info_pipe[1],&buffer,wlen) != wlen) {
-        /* Nothing to do on error, this will be detected by the other side. */
-    }
-}
-
 /* Update Child info. */
-void updateChildInfo(int information_type, size_t info) {
-    if (information_type == CHILD_INFO_TYPE_CURRENT_COW_SIZE) {
-        server.stat_current_cow_bytes = info;
-    } else if (information_type == CHILD_INFO_TYPE_CURRENT_KEYS_PROCESSED) {
-        server.stat_current_processed_keys = info;
-    } else if (information_type == CHILD_INFO_TYPE_AOF_COW_SIZE) {
-        server.stat_aof_cow_bytes = info;
-    } else if (information_type == CHILD_INFO_TYPE_RDB_COW_SIZE) {
-        server.stat_rdb_cow_bytes = info;
-    } else if (information_type == CHILD_INFO_TYPE_MODULE_COW_SIZE) {
-        server.stat_module_cow_bytes = info;
+void updateChildInfo(child_info_data *data) {
+    if (data->information_type == CHILD_INFO_TYPE_CURRENT_INFO) {
+        if (data->cow) server.stat_current_cow_bytes = data->cow;
```

Comment:
I don't like these `if`s that ignore a value of 0.
I think 0 could be a valid value. 
Instead the sender should make sure to always send valid data. 
Am I missing anything?

Reply:
I think we should be able to update only one of the two fields (keys/cow), 
(for example in the AOF diff)
maybe we should replace the `0` with `-1` to indicate no-update?

Reply:
i don't think that's necessary, but if it is, i would feel better with -1